yellowbrick/features/base.py

```diff
@@ -234,49 +325,66 @@ def fit(self, X, y=None, **kwargs):
 
-        kwargs : dict
-            Pass generic arguments to the drawing method
-
         Returns
         -------
-        self : instance
+        self : DataVisualizer
             Returns the instance of the transformer/visualizer
         """
-        super(DataVisualizer, self).fit(X, y, **kwargs)
+        # Compute the features from the data
+        super(DataVisualizer, self).fit(X, y)
 
+        # Determine the target color type
         self._determine_target_color_type(y)
 
+        # Handle the single target color type
         if self._target_color_type == TargetType.SINGLE:
-            self._colors = 'b'
+            # use the user supplied color or the first color in the color cycle
+            self._colors = self.colors or 'C0'
 
         # Compute classes and colors if target type is discrete
         elif self._target_color_type == TargetType.DISCRETE:
-            # Store the classes for the legend if they're None.
-            if self.classes_ is None:
-                # TODO: Is this the most efficient method?
-                self.classes_ = [str(label) for label in np.unique(y)]
-
-            # Ensures that classes passed by user is equal to that in target
-            if len(self.classes_)!=len(np.unique(y)):
-                warnings.warn(("Number of unique target is not "
-                              "equal to classes"), YellowbrickWarning)
-
-            color_values = resolve_colors(n_colors=len(self.classes_),
-                                          colormap=self.colormap, colors=self.color)
+            # Unique labels are used both for validation and color mapping
+            labels = np.unique(y)
+
+            # Handle user supplied classes
+            if self.classes is not None:
+                self.classes_ = np.asarray([str(c) for c in self.classes])
+
+                # Validate user supplied class labels
+                if len(self.classes_) != len(labels):
+                    raise YellowbrickValueError((
+                        "number of specified classes does not match "
+                        "number of unique values in target"
+                    ))
+
+            # Get the string labels from the unique values in y
+            else:
+                self.classes_ = np.asarray([str(c) for c in labels])
+
+            # Create a map of class labels to colors
+            color_values = resolve_colors(
+                n_colors=len(self.classes_),
+                colormap=self.colormap,
+                colors=self.colors
+            )
             self._colors = dict(zip(self.classes_, color_values))
+            self._label_encoder = dict(zip(labels, self.classes_))
 
         # Compute target range if colors are continuous
         elif self._target_color_type == TargetType.CONTINUOUS:
             y = np.asarray(y)
             self.range_ = (y.min(), y.max())
 
+            # TODO: allow for Yellowbrick palettes here as well
```

Comment:
Perhaps we could use the [updated `resolve_colors` utility](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/style/colors.py#L57-L151)?